File: src/main/java/mil/nga/tiff/field/type/ShortField.java
package mil.nga.tiff.field.type;

import java.nio.ByteBuffer;

abstract sealed class ShortField<T extends Number> extends NumericFieldType<T> permits UnsignedShortField, SignedShortField {

    @Override
    final protected void writeSample(ByteBuffer buffer, T value) {
        buffer.putShort(value.shortValue());
    }

    @Override
    final public void transferSample(ByteBuffer outBuffer, ByteBuffer inBuffer) {
        outBuffer.putShort(inBuffer.getShort());
    }

}

File: src/main/java/mil/nga/tiff/field/type/ShortField.kt
package mil.nga.tiff.field.type

import java.nio.ByteBuffer

abstract class ShortField<T : Number> : NumericFieldType<T>() {
    override fun writeSample(buffer: ByteBuffer, value: T) {
        buffer.putShort(value.toShort())
    }

    override fun transferSample(outBuffer: ByteBuffer, inBuffer: ByteBuffer) {
        outBuffer.putShort(inBuffer.getShort())
    }
}
